src/analysis/retail/evoker/augmentation/modules/features/BuffTargetHelper/BuffTargetHelper.tsx

```diff
@@ -0,0 +1,352 @@
+  ['Mage', '|cff3fc7eb'],
+  ['Paladin', '|cfff48cba'],
+  ['Warrior', '|cffc69b6d'],
+  ['Druid', '|cffff7c0a'],
+  ['DeathKnight', '|cffc41e3a'],
+  ['Hunter', '|cffaad372'],
+  ['Priest', '|cffffffff'],
+  ['Rogue', '|cfffff468'],
+  ['Shaman', '|cff0070dd'],
+  ['Warlock', '|cff8788ee'],
+  ['Monk', '|cff00ff98'],
+  ['DemonHunter', '|cffa330c9'],
+  ['Evoker', '|cff33937f'],
+]);
+
+/** SpellIds to blacklist, ie. trinkets that doesnt add contribution */
+const blacklist: number[] = [
+  402583, // Beacon
+  408671, // Bomb dispenser
+  401303, // Pocket Anvil
+  401395, // Vessel
+  418774, // Mirror
+  400223, // Thorns of Iron
+  322109, // Touch of Death
+  124280, // Touch of Karma
+  184689, // Shield of Vengeance
+];
+
+/**
+ * So managing your buffs is essentially what Augmentation boils down to.
+ * But knowing who to buff and when can be tricky to figure out at a glance.
+ * Given multiple factors such as time intervals (Ebon Might timings), filtering
+ * off damage that isn't amplified by buffs, intimate knowledge about different specs
+ * where they perform, where they don't, player skill, etc.
+ *
+ * The aim for this module is to streamline the process, by providing a list of the targets
+ * that would have been best to buff in a given time interval.
+ *
+ * Along with this we will provide the user with a MRT note export that can be used with
+ * various WA's to help keep track of who to Prescience.
+ *
+ * We don't provide analysis on actual gameplay in this module, this is strictly meant
+ * to assist people in making more informed decisions about who to buff at what times in
+ * a given boss fight.
+ *
+ * Hopefully with this information being easily available, players will be able to use this
+ * data and predict who will be high performing target on other fights as well.
+ * (Assuming you are playing with the same people again eg. reclears/prog).
+ */
+class BuffTargetHelper extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  protected combatants!: Combatants;
+
+  // This could be set to fixed 30s intervals, but might as well give some attention to the Interwoven gamers!
+  interval: number =
+    30 * 1000 * (this.selectedCombatant.hasTalent(TALENTS.INTERWOVEN_THREADS_TALENT) ? 0.9 : 1);
+
+  fightStart: number = this.owner.fight.start_time;
+  fightEnd: number = this.owner.fight.end_time;
+  mrtPrescienceHelperNote: string = '';
+
+  constructor(options: Options) {
+    super(options);
+    /** No need to show this in dungeon runs, for obvious reasons */
+    this.active = !isFightDungeon(this.owner.fight);
+
+    /** Populate our whitelist */
+    this.addEventListener(Events.fightend, () => {
+      /** Clear the maps, we do this since otherwise it has a tendency
+       * to keep the values from a previous fight if you swap around without
+       * refreshing the browser */
+      playerDamageMap.clear();
+      playerWhitelist.clear();
```

Comment:
if you move these to being class variables instead of outer variables, you shouldn't need this since the class instance will get rebuilt.